file_browser/src/classes/Folder.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  observable,
+  action,
+  ObservableMap,
+  makeObservable,
+  computed,
+} from "mobx";
+import { SystemItem } from "./SystemItem";
+import fileSystemStore from "../stores/FileSystemStore";
+
+export interface IFolder {
+  id: number;
+  name: string;
+  files: ObservableMap<number, SystemItem>;
+  parent: Folder | null;
+}
+
+export class Folder extends SystemItem implements IFolder {
+  files = observable.map<number, SystemItem>();
+
+  constructor(name: string, parent: Folder | null) {
+    super(name, parent);
+    makeObservable(this, {
+      files: observable,
+      add: action,
+      delete: action,
+      getFilteredMap: computed,
+    });
+  }
+
+  add(item: SystemItem) {
+    this.files.set(item.id, item);
+  }
+
+  delete() {
+    this.files.clear();
+    this.parent?.files.delete(this.id);
+  }
+
+  get getFilteredMap(): ObservableMap<number, SystemItem> {
```

Comment:
1. The name of the getter shouldn't start with `getX` but like the name of any property. Also, the name should represent the logical meaning of the returned content, rather than the programmatic meaning. Rename to `filteredChildren`.
2. There's no point in returning an `ObservableMap` as it will never directly change, but rather, be completely re-created. Change to a regular `Map`.